tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -11,148 +11,178 @@ using namespace shogun;
 
-	SGVector<float64_t> A(9);
-	SGVector<float64_t> B(9);
+// TODO: make global definitions
+// Definition of the 4 groups of Shogun types (shogun/mathematics/linalg/LinalgBackendBase.h)
+// TODO: add bool, chars and complex128_t types
+typedef ::testing::Types<int8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float32_t, float64_t, floatmax_t> AllTypes;
+typedef ::testing::Types<int8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t, float32_t, float64_t, floatmax_t> NonComplexTypes;
+typedef ::testing::Types<float32_t, float64_t, floatmax_t> RealTypes;
+// TODO: add complex128_t type
+typedef ::testing::Types<float32_t, float64_t, floatmax_t> NonIntegerTypes;
 
-	for (index_t i = 0; i < 9; ++i)
-	{
-		A[i] = i;
-		B[i] = 0.5*i;
-	}
+TYPED_TEST_CASE(LinalgBackendEigenAllTypesTest, AllTypes);
+TYPED_TEST_CASE(LinalgBackendEigenNonComplexTypesTest, NonComplexTypes);
+TYPED_TEST_CASE(LinalgBackendEigenRealTypesTest, RealTypes);
+TYPED_TEST_CASE(LinalgBackendEigenNonIntegerTypesTest, NonIntegerTypes);
+
+
+template <typename ST>
+void check_SGVector_add() {
+
+    const ST alpha = 1;
+    const ST beta = 2;
+
+    SGVector<ST> A(9);
+    SGVector<ST> B(9);
+
+    for (index_t i = 0; i < 9; ++i)
+    {
+        A[i] = i;
+        B[i] = 2*i;
+    }
 
-	auto result = add(A, B, alpha, beta);
+    auto result = add(A, B, alpha, beta);
 
-	for (index_t i = 0; i < 9; ++i)
-		EXPECT_NEAR(alpha*A[i]+beta*B[i], result[i], 1e-15);
+    for (index_t i = 0; i < 9; ++i)
+        EXPECT_NEAR(alpha*A[i]+beta*B[i], result[i], 1e-15);
 }
 
-TEST(LinalgBackendEigen, SGMatrix_add)
-{
-	const float64_t alpha = 0.3;
-	const float64_t beta = -1.5;
-	const index_t nrows = 2, ncols = 3;
+template <typename ST>
+void check_SGMatrix_add() {
 
-	SGMatrix<float64_t> A(nrows, ncols);
-	SGMatrix<float64_t> B(nrows, ncols);
+    const ST alpha = 1;
+    const ST beta = 2;
+    const index_t nrows = 2, ncols = 3;
 
-	for (index_t i = 0; i < nrows*ncols; ++i)
-	{
-		A[i] = i;
-		B[i] = 0.5*i;
-	}
+    SGMatrix<ST> A(nrows, ncols);
+    SGMatrix<ST> B(nrows, ncols);
 
-	auto result = add(A, B, alpha, beta);
+    for (index_t i = 0; i < nrows*ncols; ++i)
+    {
+        A[i] = i;
+        B[i] = 2*i;
+    }
 
-	for (index_t i = 0; i < nrows*ncols; ++i)
-		EXPECT_NEAR(alpha*A[i]+beta*B[i], result[i], 1e-15);
+    auto result = add(A, B, alpha, beta);
+
+    for (index_t i = 0; i < nrows*ncols; ++i)
+        EXPECT_NEAR(alpha*A[i]+beta*B[i], result[i], 1e-15);
 }
 
-TEST(LinalgBackendEigen, SGVector_add_in_place)
-{
-	const float64_t alpha = 0.3;
-	const float64_t beta = -1.5;
+template <typename ST>
+void check_SGVector_add_in_place() {
+    const ST alpha = 1;
+    const ST beta = 2;
 
-	SGVector<float64_t> A(9), B(9), C(9);
+    SGVector<ST> A(9), B(9), C(9);
 
-	for (index_t i = 0; i < 9; ++i)
-	{
-		A[i] = i;
-		B[i] = 0.5*i;
-		C[i] = i;
-	}
+    for (index_t i = 0; i < 9; ++i)
+    {
+    A[i] = i;
```

Comment:
there are some minor formatting issues here.
You an actually automatically format this properly... see the output from the first travis job here:
https://travis-ci.org/shogun-toolbox/shogun/jobs/441713579#L723

Reply:
This would also make the diff of this PR much smaller, as there are fewer whitespace changes